components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServletTest.java

```diff
@@ -0,0 +1,249 @@
+
+        when(request.getSession()).thenReturn(httpSession);
+        when(httpSession.getMaxInactiveInterval()).thenReturn(2);
+
+        mockStatic(IdentityConfigParser.class);
+        when(IdentityConfigParser.getInstance()).thenReturn(identityConfigParser);
+
+        when(request.getParameter("sessionDataKey")).thenReturn(sessionDataKey);
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oAuthServerConfiguration);
+        when(oAuthServerConfiguration.getOpenIDConnectSkipeUserConsentConfig()).thenReturn(skipUserConsent);
+
+        when(request.getParameter("id_token_hint")).thenReturn(idTokenHint);
+
+        when(OIDCSessionManagementUtil
+                .removeOPBrowserStateCookie(any(HttpServletRequest.class), any(HttpServletResponse.class)))
+                .thenReturn((Cookie) cookie);
+
+        when(OIDCSessionManagementUtil.getOIDCLogoutConsentURL()).thenReturn(redirectUrl);
+
+        when(OIDCSessionManagementUtil.getOIDCLogoutURL()).thenReturn(redirectUrl);
+
+        mockStatic(IdentityTenantUtil.class);
+        when(IdentityTenantUtil.getTenantId(anyString())).thenReturn(-1234);
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oAuthServerConfiguration);
+        when(oAuthServerConfiguration.isJWTSignedWithSPKey()).thenReturn(isJWTSignedWithSPKey);
+
+        mockStatic(KeyStoreManager.class);
+        when(KeyStoreManager.getInstance(eq(-1234))).thenReturn(keyStoreManager);
+        when(keyStoreManager.getDefaultPublicKey())
+                .thenReturn(TestUtil.getPublicKey(TestUtil.loadKeyStoreFromFileSystem(TestUtil
+                        .getFilePath("wso2carbon.jks"), "wso2carbon", "JKS"), "wso2carbon"));
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oAuthServerConfiguration);
+        when(oAuthServerConfiguration.getPersistenceProcessor()).thenReturn(tokenPersistenceProcessor);
+        when(tokenPersistenceProcessor.getProcessedClientId(anyString())).thenAnswer(new Answer<Object>() {
+
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+
+                return (String) invocation.getArguments()[0];
+            }
+        });
+
+        mockStatic(IdentityDatabaseUtil.class);
+        when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+        logoutServlet.doGet(request, response);
+        verify(response).sendRedirect(captor.capture());
+        Assert.assertTrue(captor.getValue().contains(expected));
+    }
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+
+        return new PowerMockObjectFactory();
+    }
+
+    @AfterTest
+    public void cleanData() throws Exception {
+
+        super.cleanData();
```

Comment:
what is the reason of this method. I am asking this for my understanding.

Reply:
I have used cleanData() in the parent class in order to close the db connection after test